src/ui/widgets.rs

```diff
@@ -50,6 +54,9 @@ impl<T: Spawn> Widgets for T {
                 ),
             ));
         });
+        // Add the one-shot system as a child so that will get cleaned up when the button is destroyed.
+        entity.add_child(on_press.entity());
```

Comment:
This here is the line that fixes the leak, as otherwise we never clean up the entities holding our one-shot systems but continue to register new ones when the menu is reconstructed.